goosebit/api/firmware.py

```diff
@@ -25,13 +25,27 @@ def search_filter(search_value):
         return Q(uri__icontains=search_value) | Q(version__icontains=search_value)
 
     async def parse(f):
+        compat = await f.compatibility.all().values()
+
+        result = {}
+        for entry in compat:
+            model = entry["model"]
+            revision = entry["revision"]
+            if model not in result:
+                result[model] = []
+            result[model].append(revision)
+
+        compatibility = "<br>".join([f"{model} - {', '.join(revisions)}" for model, revisions in result.items()])
```

Comment:
I think this might be better written as `dict[model, set[revision]]`.  This begs the question of restructuring the database to align with that format, namely having `Hardware` contain a `ManyToMany` field which would link to a separate model `Revision`.

Reply:
You mean returning the raw data vs. a pre-formatted string? Or simply that result[model] should not be of type list but rather set?

As for the db model: I don't see a big drawback in the current one - so hesitant to invest time there.

Reply:
I think just not a big fan of the preformatted string.  I would prefer to isolate the UI from the API as much as possible.

Reply:
Moved the formatting logic back to javascript (had it there initially)